contrib/timelapse.lua

```diff
@@ -0,0 +1,463 @@
+    ['w'] = 2560,
+    ['h'] = 1080
+  },
+  ['2K'] = {
+    ['label'] = '2K 2048x1152 (16:9)',
+    ['w'] = 2048,
+    ['h'] = 1152
+  },
+  ['4K'] = {
+    ['label'] = '4K 4096x2304 (16:9)',
+    ['w'] = 4096,
+    ['h'] = 2304
+  }
+}
+
+local framerates = {'15', '16', '23.98', '24', '25', '29,97', '30', '48', '50', '59.94', '60'}
+
+local formats = {
+  ['AVI'] = {
+    ['extension'] = 'avi',
+    ['codecs'] = {'mpeg4', 'h263p', 'h264', 'mpeg2video', 'hevc', 'vp9'}
+  },
+  ['Matroska'] = {
+    ['extension'] = 'mkv',
+    ['codecs'] = {'h264', 'h263p', 'mpeg4', 'mpeg2video', 'hevc', 'vp9'}
+  },
+  ['WebM'] = {
+    ['extension'] = 'webm',
+    ['codecs'] = {'vp9', 'h263p', 'h264', 'mpeg4', 'mpeg2video', 'hevc'}
+  },
+  ['MP4'] = {
+    ['extension'] = 'mp4',
+    ['codecs'] = {'h264', 'h263p', 'mpeg4', 'mpeg2video'}
+  },
+  ['QuickTime'] = {
+    ['extension'] = 'mov',
+    ['codecs'] = {'h264', 'h263p', 'mpeg4', 'mpeg2video'}
+  }
+}
+
+res_list = {}
+for i, v in pairs(resolutions) do
+  table.insert(res_list, v['label'])
+end
+
+table.sort(res_list)
+
+format_list = {}
+for i,v in pairs(formats) do
+  table.insert(format_list, i)
+end
+
+table.sort(format_list)
+
+codec_list = formats['AVI']['codecs']
+table.sort(codec_list)
+
+local function extract_resolution(description)
+  for _, v in pairs(resolutions) do
+    if v['label'] == description then
+      return v['w']..'x'..v['h']
+    end
+  end
+  return '100x100'
+end
+
+---- ENSURE ffmpeg is INSTALLED
+
+dt.print_log(MODULE_NAME .. " - Executable Path Preference: "..df.get_executable_path_preference("ffmpeg"))
+local ffmpeg_path = df.check_if_bin_exists("ffmpeg")
+if not ffmpeg_path then
+  dt.print_error("ffmpeg not found")
+  dt.print("ERROR - ffmpeg not found")
+end
+
+local ffmpeg_path_widget = df.executable_path_widget({'ffmpeg'})
+dt.preferences.register("executable_paths", "ffmpeg",
+  "file",
+  MODULE_NAME .. ': ffmpeg location',
+  'Install location of ffmpeg',
+  "ffmpeg",
+  ffmpeg_path_widget
+)
+
+---- GENERIC UTILS
+
+local function replace_combobox_elements(combobox, new_items, to_select)
+  if to_select == nil then
+    to_select = combobox.value
+  end
+
+  to_select_idx = 1
+  for i , name in ipairs(new_items) do
+    if name == to_select then
+      to_select_idx = i
+      break
+    end
+  end
+
+  local old_elements_count = #combobox
+  for i, name in ipairs(new_items) do
+    combobox[i] = name
+  end
+  if old_elements_count > #new_items then
+    for j = old_elements_count, #new_items + 1, -1 do
+      combobox[j] = nil
+    end
+  end
+
+  combobox.value = to_select_idx
+end
+
+local function format(label, symbols)
+  local es1, es2 = "\u{ffe0}", "\u{ffe1}" -- for simplicity, just some strange utf characters 
+  local result = label:gsub("\\{", es1):gsub("\\}", es2)
+  for s,v in pairs(symbols) do
+    result = result:gsub("{"..s.."}", v)
+  end
+  return result:gsub(es1, "{"):gsub(es2, "}")
+end
+
+local function open_file(filename) 
```

Comment:
could we change this to something like launch_default_app?  open_file makes it sound like you're opening a file for  reading or writing, not launching an application that's associated with the file.

@supertobi this could probably end up in the system library.

Reply:
renamed & moved to system lib